locker

```diff
@@ -7,41 +7,72 @@ function lock_image() {
 	echo $NAME@$DIGEST
 }
 
-function lock_dockerfile() {
-	FILE=$1
+function unlock_image() {
+	IMAGE=$1
+	NAME=$(echo $IMAGE | grep '^[^@]\+' -o)
+	echo $NAME
+}
+
+function update_dockerfile() {
+	OPERATION=$1
+	FILE=$2
 	LOCKFILE=$FILE.lock
 
 	while IFS= read -r LINE; do
-		if [[ $LINE =~ ^FROM ]]
+		if [[ $LINE =~ ^FROM ]] && [[ $OPERATION == "lock" ]]
 		then
 			IMAGE=$(echo "$LINE" | sed -r 's/^FROM +([^ ]+).*$/\1/')
 			LOCKIMAGE=$(lock_image $IMAGE)
 			echo "$LINE" | sed -r "s|$IMAGE|$LOCKIMAGE|" >>$LOCKFILE
+		elif [[ $LINE =~ ^FROM ]] && [[ $OPERATION == "unlock" ]]
+		then
+			IMAGE=$(echo "$LINE" | sed -r 's/^FROM +([^ ]+).*$/\1/')
+			LOCKIMAGE=$(unlock_image $IMAGE)
+			echo "$LINE" | sed -r "s|$IMAGE|$LOCKIMAGE|" >>$LOCKFILE
```

Comment:
A lot of duplication here. Be good to keep it as before but somehow pass the operation as a function through? Not sure how best to do that in Bash without some reading.

Reply:
Thinking of the flow as:
* Main switch matches on file type (e.g. Dockerfile, docker-compose.yml)
* Dockerfile function pulls out images for the operation (and a future docker-compose function does the same but for YAML)
* Operation locks/unlocks (irrespective of file type)
Future further outer loop would scan directory to process files via the above.

Reply:
Yeah I wondered about this too. The current way was just to avoid ugly nesting. Maybe there is a smart way to do this? I can have a look.

Reply:
Not sure how I feel about my solution to this... Let me know what you think, reduces the duplication at least 🤷‍♂

Reply:
Thought some kind of eval would work :) It's better than duplication for sure.